### DISCUSSION TITLE
tooltip forwardRef 的疑问

### DISCUSSION THREAD

**Discussion by theFG1998** [ant-design/ant-design#42123] [Q&A]
```typescript
const Tooltip = React.forwardRef<TooltipRef, TooltipProps>((props, ref) => {
  const {
    prefixCls: customizePrefixCls,
    openClassName,
    getTooltipContainer,
    overlayClassName,
    color,
    overlayInnerStyle,
    children,
    afterOpenChange,
    afterVisibleChange,
    destroyTooltipOnHide,
    arrow = true,
    title,
    overlay,
    builtinPlacements,
    arrowPointAtCenter = false,
    autoAdjustOverflow = true,
  } = props;

  const mergedShowArrow = !!arrow;

  const { token } = useToken();

  const {
    getPopupContainer: getContextPopupContainer,
    getPrefixCls,
    direction,
  } = React.useContext(ConfigContext);

  // ============================== Ref ===============================
  const tooltipRef = React.useRef<RcTooltipRef>(null);

  const forceAlign = () => {
    tooltipRef.current?.forceAlign();
  };
....
  ref={tooltipRef}
```
实际传给内层组件的不是forwardRef的第二个参数而是forwardRef函数内部新建的tooltipRef，请问这样的意义是什么，forwardRef是不是等于失效了

**Comment by zombieJ** [comment]
可以看一下 [useImperativeHandle](https://react.dev/reference/react/useImperativeHandle) hooks 的介绍，`ref` 是可以代理的。

https://github.com/ant-design/ant-design/blob/c3d02f82120de719d010e73ddcf56677b1be06ac/components/tooltip/index.tsx#L227

**Reply by theFG1998** [reply]
> 可以看一下 [useImperativeHandle](https://react.dev/reference/react/useImperativeHandle) hooks 的介绍，`ref` 是可以代理的。
> 
> https://github.com/ant-design/ant-design/blob/c3d02f82120de719d010e73ddcf56677b1be06ac/components/tooltip/index.tsx#L227

但这边有个需求需要拿到tooltip内部组件的Ref，请问这种情况该咋整
